### PR TITLE
Add ability to choose a default authentication flow for a provider

### LINKED ISSUE #3694: Add ability to choose a default authentication flow for a provider
**Is your feature request related to a problem? Please describe.**

Currently, choosing a default authentication flow for a provider/app involves monkeying around with policies to make sure that only one of them passes for said provider/app.

**Describe the solution you'd like**

As it is done for authorization flows, add an option to choose an authentication flow for a provider.

**Describe alternatives you've considered**

No alternative have been discussed yet.

**Additional context**

This feature was discussed on [Discord](https://discordapp.com/channels/809154715984199690/809154716507963434/1024684056979456000).

### PR DESCRIPTION
**Pull request by rissson**
# Details
* **Does this resolve an issue?**
Resolves #3694

### New Features
* Add ability to choose a default authentication flow for a provider

## Additional

Questions I ask myself:

* Does it make sense to have this for LDAP/Proxy providers?
* Is it okay not to specify a default authentication flow? My guess is yes, as it will fallback on the current behavior.

### CODE REVIEW

File: authentik/core/models.py [hunk 1]
```diff
@@ -258,6 +258,15 @@ class Provider(SerializerModel):
 
     name = models.TextField()
 
+    authentication_flow = models.ForeignKey(
+        "authentik_flows.Flow",
+        null=True,
+        blank=True,
```
Comment (BeryJu) [inline authentik/core/models.py:264]:
```suggestion
```

File: authentik/core/models.py [hunk 2]
```diff
@@ -258,6 +258,15 @@ class Provider(SerializerModel):
...
+        on_delete=models.CASCADE,
```
Comment (BeryJu) [inline authentik/core/models.py:265]:
```suggestion
        on_delete=models.SET_NULL,
        default=None,
```

**Review by BeryJu — CHANGES_REQUESTED** [review]
mostly LGTM, the only changes requested are for phrasing, and changing some database constraints

**Comment by Zegorax** [comment]
@BeryJu What is missing in this PR in order to have it merged ?

**Comment by rissson** [comment]
I still haven't applied his suggestions. I'll get around to it soon.

**Comment by Zegorax** [comment]
@rissson Do you need any help to finish it?

**Comment by rissson** [comment]
@BeryJu I applied your suggestions. Would it be possible to re-open this?

**Comment by BeryJu** [comment]
image
Github actually doesn't let me, but while I'm at it I'll disable stalebot for PRs as it doesn't really make sense

**Comment by BeryJu** [comment]
You should be able to create a new PR with the same branch though

**Pull request closed without merging**